src/unix/bsd/netbsdlike/netbsd/riscv64.rs

```diff
@@ -3,11 +3,72 @@ use PT_FIRSTMACH;
 pub type c_long = i64;
 pub type c_ulong = u64;
 pub type c_char = u8;
+pub type __greg_t = u64;
 pub type __cpu_simple_lock_nv_t = ::c_int;
 
+s! {
+    pub struct mcontext_t {
+        pub __gregs: [::greg_t; 32],
+        pub __fregs: [__c_anonymous__fpreg; 33],
+        __spare: [::__greg_t; 7],
+    }
+}
+
+s_no_extra_traits! {
+    #[cfg_attr(feature = "extra_traits", allow(missing_debug_implementations))]
+    pub union __c_anonymous__fpreg {
+        pub u_u64: u64,
+        pub u_d: ::c_double,
+    }
```

Comment:
Sorry, one other thing - should this just be called `__fpreg`? From https://github.com/NetBSD/src/blob/0465e5c825effb130eca95499f5ac6454fe0d5ab/sys/arch/riscv/include/mcontext.h#L44C7-L47.

It would probably also be good to define 

```rust
pub const _NGREG: usize = 32;
pub const _NFREG: usize = 33;
pub type __gregset_t = [__greg_t; _NGREG];
pub type __fregset_t = [__fpreg; _NFREG];
```

and then use those in mcontext